internal/controllers/phase_ensure_wanted_namespaces.go

```diff
@@ -78,6 +77,12 @@ func (r *AddonReconciler) ensureWantedNamespaces(
 }
 
 // Ensure a single Namespace for the given Addon resource
+// TODO refactor ideas
+// - create desired ns object
+// - get current ns from apiserver, if dont exist, create desired ns
+// - if exists, try to SetControllerRef + check for AlreadyOwnedError
+// - if success, take ownership + set commonLabels + Update current ns
+// - if fails, report AlreadyOwnedError (replaces errNotOwnedByUs)
```

Comment:
Sounds like a plan, would love to see that moved to it's own PR. :)

Reply:
me too :)